gcpauth/token.go

```diff
@@ -0,0 +1,71 @@
+package gcpauth
+
+import (
+	"context"
+	"os"
+
+	credentials "cloud.google.com/go/iam/credentials/apiv1"
+	"google.golang.org/api/option"
+	credentialsPb "google.golang.org/genproto/googleapis/iam/credentials/v1"
+)
+
+const (
+	EnvServiceAcctFile = "GOOGLE_APPLICATION_CREDENTIALS"
+)
+
+func isEnvExist(key string) bool {
+	_, ok := os.LookupEnv(key)
+	return ok
+}
+
+//GetAuthToken returns an access token from GCP
+//If the GOOGLE_APPLICATION_CREDENTIALS environment variable is set, it will read an auth.json file from the path
+//If it isn't set, it will use the use internal GCP mechanism to authenticate it's self.
+func GetAuthToken(saEmail string) (string, error) {
+	if isEnvExist(EnvServiceAcctFile) {
+		return GetAuthFromFile(os.Getenv(EnvServiceAcctFile), saEmail)
+	}
+	return GetAuthFromKube(saEmail)
+}
+
+func GetAuthFromKube(saEmail string) (string, error) {
+	ctx := context.Background()
+	credentialsClient, err := credentials.NewIamCredentialsClient(ctx)
+	if err != nil {
+		return "", err
+	}
+	defer credentialsClient.Close()
+
+	return getToken(ctx, saEmail, credentialsClient)
+}
+
+func GetAuthFromFile(path, saEmail string) (string, error) {
+	ctx := context.Background()
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		return "", err
+	} else if err != nil {
+		return "", err
+	}
```

Comment:
Just return the error directly. `os.IsNotExist(err)` is not serving any purpose here.